kura/org.eclipse.kura.core.keystore/OSGI-INF/metatype/org.eclipse.kura.core.keystore.FilesystemKeystoreServiceImpl.xml

```diff
@@ -45,8 +45,88 @@
             description="Specifies if the defined password will be randomized at the next keystore access. If this value is set to true and the keystore can be accessed, the password will be randomized and this field will automatically set to false.">
         </AD>
 
+        <AD id="crl.management.enabled"
+            name="CRL Cache Enabled"
+            type="Boolean"
+            cardinality="0"
+            required="true"
+            default="false"
+            description="Enables or disables CRL cache.">
+        </AD>
+
+        <AD id="crl.update.interval"
+            name="CRL Force Update Interval"
+            type="Long"
+            cardinality="0"
+            required="true"
+            default="1"
+            description="Defines a time interval for forcing the update of cached CRLs, this interval will be considered in addition to CRL next update date.">
+        </AD>
+
+        <AD id="crl.update.interval.time.unit"
+            name="CRL Force Update Interval Time Unit"
+            type="String"
+            cardinality="0"
+            required="true"
+            default="DAYS"
+            description="The time unit for the CRL Update Interval parameter">
+            <Option label="SECONDS" value="SECONDS" />
+            <Option label="MINUTES" value="MINUTES" />
+            <Option label="HOURS" value="HOURS" />
+            <Option label="DAYS" value="DAYS" />
+        </AD>
+
+        <AD id="crl.check.interval"
+            name="CRL Check Interval"
+            type="Long"
+            cardinality="0"
+            required="true"
+            default="5"
+            description="Defines a time interval for the periodic check of stored CRLs.">
+        </AD>
+
+        <AD id="crl.check.interval.time.unit"
+            name="CRL Check Interval Time Unit"
+            type="String"
+            cardinality="0"
+            required="true"
+            default="MINUTES"
+            description="The time unit for the CRL Check Interval parameter">
+            <Option label="SECONDS" value="SECONDS" />
+            <Option label="MINUTES" value="MINUTES" />
+            <Option label="HOURS" value="HOURS" />
+            <Option label="DAYS" value="DAYS" />
+        </AD>
+
+        <AD id="crl.store.path"
+            name="CRL Store Path"
+            type="String"
+            cardinality="0"
+            required="false"
+            default=""
+            description="Defines the path of the CRL store file.">
+        </AD>
+
+        <AD id="verify.crl"
+            name="Enable CRL Verification"
+            type="Boolean"
+            cardinality="0"
+            required="true"
+            default="true"
+            description="If set to true, the downloaded CRLs will be stored only if signed with the public key of one of trusted certificates in this keystore.">
+        </AD>
+
+        <AD id="crl.urls"
```

Comment:
If URLs may be needed for the feature to work, maybe the field should be put on top after the enabling